### ISSUE TITLE
argument destructuring breaks several test functions

### ISSUE THREAD

**Issue by machow** [datacamp/pythonwhat#141]
For solution code like,

```{python}
[i for i,*j in zip([1,2],[3,4], [5,6])]
```

trying to grade it with a `test_list_comp` will throw an error:

```
DataCamp encountered the following error:
'Starred' object has no attribute 'id'
```

 This is a quick fix in the parser code. However,

```{python}
class A:
    def __enter__(self): return 1, 2, 3
    def __exit__(self, *args, **kwargs): print('exited')

with A() as (a, *b): 
    pass
```

will still throw an error when it tries to setup the context manager in each process, since it matches each target variable manually to the output. A helper function to do destructuring assignment by exec'ing ast targets/optional_vars should fix the problem.

**Comment by machow** [comment]
Will be fixed by 81d4915b8. Note that context_vals don't currently use AST magic to assign values like `with_context`.

**Comment by machow** [comment]
merged into master
